### PR TITLE
Restricting spacy.cli for version 3.7.0

### PR DESCRIPTION
**Pull request by kshitijcode**
## Change Description

Describe your changes

Puts restriction on installing spacy 3.7.0

## Issue reference

This PR fixes issue https://github.com/microsoft/presidio/issues/1487

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by omri374** [comment]
Thanks! Do you think it would be possible to do this for the presidio-analyzer module? This way it would avoid this pitfall for other use cases as well, not just for CLI users

**Comment by kshitijcode** [comment]
@omri374 added it to the presidio-analyzer module as well.

**Comment by SharonHart** [comment]
Isn't defining it in the upper-most pyptoject.toml making it apply to all modules?

File: presidio-analyzer/pyproject.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ azure-ai-textanalytics = { version = "*", optional = true }
 azure-core = { version = "*", optional = true }
 transformers = { version = "*", optional = true }
 huggingface_hub = { version = "*", optional = true }
+spacy = "!=3.7.0"  # Exclude spaCy 3.7.0 -> https://github.com/microsoft/presidio/issues/1487
```
Comment (omri374) [inline presidio-analyzer/pyproject.toml:39]: spacy is already on line 26. Could you please add the 3.7.0 constraint to this line? e.g. `>=3.4.4, <4.0.0, !=3.7.0`

**Review by omri374 — APPROVED** [review]

**Pull request merged**